### PR TITLE
fix(datasource-sql): don't return function names when default values are not constants

### PR DESCRIPTION
**Pull request by ghusse**
Linked to CU-860rwux0k

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Comment by realSpok** [comment]
I did not manage to use a string default value with mysql.
example of column schema:

```sql
 `default_str` varchar(100) DEFAULT 'abc',
```

Intropsection works well but at Sequelize crashes at creation. I believe it is trying to insert 'abc' as column name

Store creation failed: SequelizeDatabaseError: (conn=44, no: 1054, SQLState: 42S22) Unknown column 'abc' in 'field list' sql: INSERT INTO `store` (`id`,`name`,`owner_id`,`fn`,`default_str`) VALUES (NULL,'aaabcc',200,curdate(),abc); - parameters:[]

File: packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        AND ${conditions}`;
+
+    const replacements = tableNames.reduce(
+      (acc, tableName, index) => ({
+        ...acc,
+        [`tableName${index}`]: tableName.tableName,
+        [`schemaName${index}`]: tableName.schema || 'dbo',
+      }),
+      {
+        database: sequelize.getDatabaseName(),
```
Comment (realSpok) [inline packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts:83]: This doesn't work if you don't pass the database name explicitely:
ex:
```
     createSqlDataSource('mysql://example:password@localhost:3306'),
```



On `main` it was also crashing, but with a different error
```
  name: 'SequelizeDatabaseError',
  parent: Error: No database selected
      at Packet.asError (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/packets/packet.js:728:17)
      at Query.execute (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/commands/command.js:29:26)
      at Connection.handlePacket (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/connection.js:478:34)
      at PacketParser.onPacket (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/connection.js:97:12)
      at PacketParser.executeStart (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/packet_parser.js:75:16)
      at Socket.<anonymous> (/Users/nicolas/Forest/agent-nodejs/node_modules/mysql2/lib/connection.js:104:25)
      at Socket.emit (node:events:513:28)
      at addChunk (node:internal/streams/readable:324:12)
      at readableAddChunk (node:internal/streams/readable:297:9)
      at Readable.push (node:internal/streams/readable:234:10) {
    code: 'ER_NO_DB_ERROR',
    errno: 1046,
    sqlState: '3D000',
    sqlMessage: 'No database selected',
    sql: 'SHOW FULL COLUMNS FROM `store`;',
    parameters: undefined
  },
```

Comment (realSpok) [reply packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts:83]: Not sure if it is worth a fix,

File: packages/datasource-sql/src/introspection/dialects/mysql-dialect/parse-enum.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
+export default function parseEnum(type: string | null): string[] {
```
Comment (realSpok) [inline packages/datasource-sql/src/introspection/dialects/mysql-dialect/parse-enum.ts:1]:
```suggestion
/**
 * "ENUM('enum1','enum2')" returns ['enum1', 'enum2']]
 */
export default function parseEnum(type: string | null): string[] {
```

File: packages/datasource-sql/test/introspection/dialects/postgresql/parse-array.test.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import parseArray from '../../../../src/introspection/dialects/postgresql-dialect/parse-array';
```
Comment (realSpok) [inline packages/datasource-sql/test/introspection/dialects/postgresql/parse-array.test.ts:1]: should be at `packages/datasource-sql/test/introspection/dialects/postgresql-dialect/parse-array.test.ts`

File: packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    }
+
+    const defaultValueWithoutParenthesis = description.defaultValue.slice(1, -1);
+
+    if (
+      defaultValueWithoutParenthesis.startsWith("N'") &&
+      defaultValueWithoutParenthesis.endsWith("'")
+    ) {
+      return {
+        defaultValue: this.extractConstantValue(defaultValueWithoutParenthesis.slice(2, -1)),
```
[the remark this answers is not in the document]
Comment (ghusse) [reply packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts:158]: It's a little faster I think. Here the expression is very simple so we don't need to use regex.

The little benchmark I prepared shows that the regex version is 77% slower: https://jsbench.me/b2lp8f7wzs/1

File: packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts (hunk 1 above)
Comment (ghusse) [reply packages/datasource-sql/src/introspection/dialects/mssql-dialect.ts:85]: Fixed, thanks

File: packages/datasource-sql/test/introspection/dialects/postgresql/parse-array.test.ts (hunk 3 above)
Comment (ghusse) [reply packages/datasource-sql/test/introspection/dialects/postgresql/parse-array.test.ts]: I renamed the directories, thanks

**Review by realSpok — APPROVED** [review]
- ✅  Code review
  - ✅  Use pure functions when possible
  - ✅  Performance concerns
  - ✅  Security concerns
  - ✅  The PR tackle one subject only
- ✅ Automatic tests
  - ✅  Unit tests
  - ✅  Integration tests
- ✅ Manual tests
  -  ✅ Test the functionality
  -  ✅  Test error cases
 ✅   PR linked to the clickup task -> don't forget to update the community post once released
- ✅   PR title
- ✅   The PR tackles only one subject

**Pull request merged**